### ISSUE TITLE
Problem in clinicalData function in For loops

### ISSUE THREAD

**Issue by canholyavkin** [waldronlab/cBioPortalData#18]
I was using the **cBioPortalData** r package to extract the clinical data of the several studies. However, the clinicalData function doesn't work in for loop and gives only the first record. Here is reproducible example.

```r
library(cBioPortalData)

cbio <- cBioPortal()

stud <- c("paac_jhu_2014", "mel_tsam_liang_2017", "all_stjude_2015")

for (i in 1:3) {
  print(clinicalData(cbio, studyId = stud[i]))
}
```
It only extracts the clinical data of the first study (paac_jhu_2014) and ignores the rest of them.

```
# A tibble: 23 x 9
   uniquePatientKey              patientId studyId       AGE   DIAGNOSIS                           ETHNICITY SAMPLE_COUNT SEX   STAGE
   <chr>                         <chr>     <chr>         <chr> <chr>                               <chr>     <chr>        <chr> <chr>
 1 QUNJTkFSMDE6cGFhY19qaHVfMjAx… ACINAR01  paac_jhu_2014 53    Acinar cell carcinoma               White     1            Male  1B   
 2 QUNJTkFSMDI6cGFhY19qaHVfMjAx… ACINAR02  paac_jhu_2014 77    Acinar cell carcinoma               White     1            Male  2B   
 3 QUNJTkFSMDM6cGFhY19qaHVfMjAx… ACINAR03  paac_jhu_2014 60    Acinar cell carcinoma               White     1            Male  2B   
 4 QUNJTkFSMDQ6cGFhY19qaHVfMjAx… ACINAR04  paac_jhu_2014 73    Mixed Acinar-Ductal Carcinoma       White     1            Fema… 2B   
 5 QUNJTkFSMDU6cGFhY19qaHVfMjAx… ACINAR05  paac_jhu_2014 56    Acinar cell carcinoma               White     1            Male  2A   
 6 QUNJTkFSMDY6cGFhY19qaHVfMjAx… ACINAR06  paac_jhu_2014 55    Acinar cell carcinoma               White     1            Fema… 1B   
 7 QUNJTkFSMDg6cGFhY19qaHVfMjAx… ACINAR08  paac_jhu_2014 19    Mixed Acinar-Neuroendocrine Carcin… White     1            Fema… 1B   
 8 QUNJTkFSMDk6cGFhY19qaHVfMjAx… ACINAR09  paac_jhu_2014 37    Acinar Cell Carcinoma               White     1            Fema… 2A   
 9 QUNJTkFSMTA6cGFhY19qaHVfMjAx… ACINAR10  paac_jhu_2014 52    Mixed Acinar-Ductal Carcinoma       Unknown   1            Male  2B   
10 QUNJTkFSMTE6cGFhY19qaHVfMjAx… ACINAR11  paac_jhu_2014 66    Acinar Cell Carcinoma               Unknown   1            Male  2A   
# … with 13 more rows
# A tibble: 23 x 9
   uniquePatientKey              patientId studyId       AGE   DIAGNOSIS                           ETHNICITY SAMPLE_COUNT SEX   STAGE
   <chr>                         <chr>     <chr>         <chr> <chr>                               <chr>     <chr>        <chr> <chr>
 1 QUNJTkFSMDE6cGFhY19qaHVfMjAx… ACINAR01  paac_jhu_2014 53    Acinar cell carcinoma               White     1            Male  1B   
 2 QUNJTkFSMDI6cGFhY19qaHVfMjAx… ACINAR02  paac_jhu_2014 77    Acinar cell carcinoma               White     1            Male  2B   
 3 QUNJTkFSMDM6cGFhY19qaHVfMjAx… ACINAR03  paac_jhu_2014 60    Acinar cell carcinoma               White     1            Male  2B   
 4 QUNJTkFSMDQ6cGFhY19qaHVfMjAx… ACINAR04  paac_jhu_2014 73    Mixed Acinar-Ductal Carcinoma       White     1            Fema… 2B   
 5 QUNJTkFSMDU6cGFhY19qaHVfMjAx… ACINAR05  paac_jhu_2014 56    Acinar cell carcinoma               White     1            Male  2A   
 6 QUNJTkFSMDY6cGFhY19qaHVfMjAx… ACINAR06  paac_jhu_2014 55    Acinar cell carcinoma               White     1            Fema… 1B   
 7 QUNJTkFSMDg6cGFhY19qaHVfMjAx… ACINAR08  paac_jhu_2014 19    Mixed Acinar-Neuroendocrine Carcin… White     1            Fema… 1B   
 8 QUNJTkFSMDk6cGFhY19qaHVfMjAx… ACINAR09  paac_jhu_2014 37    Acinar Cell Carcinoma               White     1            Fema… 2A   
 9 QUNJTkFSMTA6cGFhY19qaHVfMjAx… ACINAR10  paac_jhu_2014 52    Mixed Acinar-Ductal Carcinoma       Unknown   1            Male  2B   
10 QUNJTkFSMTE6cGFhY19qaHVfMjAx… ACINAR11  paac_jhu_2014 66    Acinar Cell Carcinoma               Unknown   1            Male  2A   
# … with 13 more rows
# A tibble: 23 x 9
   uniquePatientKey              patientId studyId       AGE   DIAGNOSIS                           ETHNICITY SAMPLE_COUNT SEX   STAGE
   <chr>                         <chr>     <chr>         <chr> <chr>                               <chr>     <chr>        <chr> <chr>
 1 QUNJTkFSMDE6cGFhY19qaHVfMjAx… ACINAR01  paac_jhu_2014 53    Acinar cell carcinoma               White     1            Male  1B   
 2 QUNJTkFSMDI6cGFhY19qaHVfMjAx… ACINAR02  paac_jhu_2014 77    Acinar cell carcinoma               White     1            Male  2B   
 3 QUNJTkFSMDM6cGFhY19qaHVfMjAx… ACINAR03  paac_jhu_2014 60    Acinar cell carcinoma               White     1            Male  2B   
 4 QUNJTkFSMDQ6cGFhY19qaHVfMjAx… ACINAR04  paac_jhu_2014 73    Mixed Acinar-Ductal Carcinoma       White     1            Fema… 2B   
 5 QUNJTkFSMDU6cGFhY19qaHVfMjAx… ACINAR05  paac_jhu_2014 56    Acinar cell carcinoma               White     1            Male  2A   
 6 QUNJTkFSMDY6cGFhY19qaHVfMjAx… ACINAR06  paac_jhu_2014 55    Acinar cell carcinoma               White     1            Fema… 1B   
 7 QUNJTkFSMDg6cGFhY19qaHVfMjAx… ACINAR08  paac_jhu_2014 19    Mixed Acinar-Neuroendocrine Carcin… White     1            Fema… 1B   
 8 QUNJTkFSMDk6cGFhY19qaHVfMjAx… ACINAR09  paac_jhu_2014 37    Acinar Cell Carcinoma               White     1            Fema… 2A   
 9 QUNJTkFSMTA6cGFhY19qaHVfMjAx… ACINAR10  paac_jhu_2014 52    Mixed Acinar-Ductal Carcinoma       Unknown   1            Male  2B   
10 QUNJTkFSMTE6cGFhY19qaHVfMjAx… ACINAR11  paac_jhu_2014 66    Acinar Cell Carcinoma               Unknown   1            Male  2A   
# … with 13 more rows
```

I get similar results even I use `lapply` instead of for loop.

```r
lapply(stud, function(x) clinicalData(cbio, x))
```

Could you please help me in this issue?

Thanks

**Comment by LiNk-NY** [comment]
Hi Can, @canholyavkin 
Thanks for reporting this. I am aware that there is an issue with
the caching that has cropped up recently. I've been investigating
and fixing. 

What version of R, Bioconductor, and package are you using?
Thanks.
-Marcel

**Comment by canholyavkin** [comment]
Thank you for the reply.

```
R version 4.0.0
Bioconductor version 3.11
cBioPortal version 2.0.4
```

**Comment by LiNk-NY** [comment]
@canholyavkin Thanks for the report.
It should be fixed in release a97fa9d and devel b8226b9.

**Comment by canholyavkin** [comment]
Yes, the problem is solved now. Thanks.

```
devtools::install_github("waldronlab/cBioPortalData")

library(cBioPortalData)

cbio <- cBioPortal()
stud <- c("paac_jhu_2014", "mel_tsam_liang_2017", "all_stjude_2015")

for (i in 1:3) {
  print(clinicalData(cbio, studyId = stud[i]))
}
```

```
# A tibble: 23 x 9
   uniquePatientKey              patientId studyId       AGE   DIAGNOSIS                           ETHNICITY SAMPLE_COUNT SEX   STAGE
   <chr>                         <chr>     <chr>         <chr> <chr>                               <chr>     <chr>        <chr> <chr>
 1 QUNJTkFSMDE6cGFhY19qaHVfMjAx… ACINAR01  paac_jhu_2014 53    Acinar cell carcinoma               White     1            Male  1B   
 2 QUNJTkFSMDI6cGFhY19qaHVfMjAx… ACINAR02  paac_jhu_2014 77    Acinar cell carcinoma               White     1            Male  2B   
 3 QUNJTkFSMDM6cGFhY19qaHVfMjAx… ACINAR03  paac_jhu_2014 60    Acinar cell carcinoma               White     1            Male  2B   
 4 QUNJTkFSMDQ6cGFhY19qaHVfMjAx… ACINAR04  paac_jhu_2014 73    Mixed Acinar-Ductal Carcinoma       White     1            Fema… 2B   
 5 QUNJTkFSMDU6cGFhY19qaHVfMjAx… ACINAR05  paac_jhu_2014 56    Acinar cell carcinoma               White     1            Male  2A   
 6 QUNJTkFSMDY6cGFhY19qaHVfMjAx… ACINAR06  paac_jhu_2014 55    Acinar cell carcinoma               White     1            Fema… 1B   
 7 QUNJTkFSMDg6cGFhY19qaHVfMjAx… ACINAR08  paac_jhu_2014 19    Mixed Acinar-Neuroendocrine Carcin… White     1            Fema… 1B   
 8 QUNJTkFSMDk6cGFhY19qaHVfMjAx… ACINAR09  paac_jhu_2014 37    Acinar Cell Carcinoma               White     1            Fema… 2A   
 9 QUNJTkFSMTA6cGFhY19qaHVfMjAx… ACINAR10  paac_jhu_2014 52    Mixed Acinar-Ductal Carcinoma       Unknown   1            Male  2B   
10 QUNJTkFSMTE6cGFhY19qaHVfMjAx… ACINAR11  paac_jhu_2014 66    Acinar Cell Carcinoma               Unknown   1            Male  2A   
# … with 13 more rows
# A tibble: 34 x 24
   uniquePatientKey patientId studyId AGE_AT_DIAGNOSIS DFS_MONTHS DFS_STATUS FOLLOW_UP_TIME_… IN_TRANSIT KBASE_ID LYMPH_NODE_EXAM…
   <chr>            <chr>     <chr>   <chr>            <chr>      <chr>      <chr>            <chr>      <chr>    <chr>           
 1 UC0xOm1lbF90c2F… P-1       mel_ts… 77               21         Recurred/… 28               No         1        0               
 2 UC0yOm1lbF90c2F… P-2       mel_ts… 67               29.1       Recurred/… 60.3             No         2        2               
 3 UC0zOm1lbF90c2F… P-3       mel_ts… 77               13.3       Recurred/… 41               Yes        3        1               
 4 UC00Om1lbF90c2F… P-4       mel_ts… 39               5.5        Recurred/… 34               No         4        9               
 5 UC01Om1lbF90c2F… P-5       mel_ts… 47               32.6       Recurred/… 101.4            Yes        5        3               
 6 UC02Om1lbF90c2F… P-6       mel_ts… 56               10.7       DiseaseFr… 45.6             No         6        2               
 7 UC03Om1lbF90c2F… P-7       mel_ts… 46               51.2       Recurred/… 85.9             No         7        1               
 8 UC04Om1lbF90c2F… P-8       mel_ts… 59               10.8       Recurred/… 13.4             No         8        2               
 9 UC05Om1lbF90c2F… P-9       mel_ts… 86               9.8        Recurred/… 15.4             No         9        NA              
10 UC0xMDptZWxfdHN… P-10      mel_ts… 78               9.6        Recurred/… 19.2             Yes        10       0               
# … with 24 more rows, and 14 more variables: METASTASIS <chr>, OR_TIME <chr>, OS_MONTHS <chr>, OS_STATUS <chr>,
#   PRIMARY_MELANOMA_TUMOR_ULCERATION <chr>, RACE <chr>, RADIATION_TREATMENT_ADJUVANT <chr>, SAMPLE_COUNT <chr>, SEX <chr>,
#   SITE_FIRST_RECURRENCE <chr>, TIME_TO_RECURRENCE_MONTHS <chr>, TREATMENT <chr>, ALTERNATE_THERAPY_OTHER <chr>,
#   OTHER_TREATMENT_RESPONSE <chr>
# A tibble: 85 x 7
   uniquePatientKey                 patientId studyId         AGE   COHORT             SAMPLE_COUNT SEX   
   <chr>                            <chr>     <chr>           <chr> <chr>              <chr>        <chr> 
 1 U0pJTkYwMDE6YWxsX3N0anVkZV8yMDE1 SJINF001  all_stjude_2015 0.17  INF_MLLr_Discovery 2            Female
 2 U0pJTkYwMDI6YWxsX3N0anVkZV8yMDE1 SJINF002  all_stjude_2015 0.58  INF_MLLr_Discovery 2            Male  
 3 U0pJTkYwMDM6YWxsX3N0anVkZV8yMDE1 SJINF003  all_stjude_2015 0.58  INF_MLLr_Discovery 1            Male  
 4 U0pJTkYwMDQ6YWxsX3N0anVkZV8yMDE1 SJINF004  all_stjude_2015 0.58  INF_MLLr_Discovery 1            Female
 5 U0pJTkYwMDU6YWxsX3N0anVkZV8yMDE1 SJINF005  all_stjude_2015 1     INF_MLLr_Discovery 1            Female
 6 U0pJTkYwMDY6YWxsX3N0anVkZV8yMDE1 SJINF006  all_stjude_2015 0.17  INF_MLLr_Discovery 1            Male  
 7 U0pJTkYwMDc6YWxsX3N0anVkZV8yMDE1 SJINF007  all_stjude_2015 0.33  INF_MLLr_Discovery 1            Male  
 8 U0pJTkYwMDg6YWxsX3N0anVkZV8yMDE1 SJINF008  all_stjude_2015 0.33  INF_MLLr_Discovery 1            Male  
 9 U0pJTkYwMDk6YWxsX3N0anVkZV8yMDE1 SJINF009  all_stjude_2015 0.02  INF_MLLr_Discovery 1            Male  
10 U0pJTkYwMTA6YWxsX3N0anVkZV8yMDE1 SJINF010  all_stjude_2015 0.04  INF_MLLr_Discovery 1            Male  
# … with 75 more rows
```

**Comment by canholyavkin** [comment]
Just a small feedback. lapply based function still brings the first cases even after the update.

```
lapply(stud, function(x) clinicalData(cbio, x))
```

**Comment by LiNk-NY** [comment]
Thanks for the note. Fixed in 76c00cd88750da18cba945e1347d2d4a39244e1a
